docs/setup_guide.md

```diff
@@ -0,0 +1,120 @@
+
+### Enabling the module
+
+To allow the DAO module to actually execute transaction it is required to enable it on the Safe that it is connected to. For this it is possible to use the Transaction Builder on https://app.gnosis-safe.io. For this you can follow our tutorial on [adding a module](https://help.gnosis-safe.io/en/articles/4934427-add-a-module).
+
+## Snapshot integration
+
+Once the module is setup it is possible to configure a space on [Snapshot](https://snapshot.org/) to enable the DAO module plugin. For this the space configuration needs to include `"plugins": { "daoModule": { "address": "<module_address"} }`. An example for this can be found in the [🍯DAO space configuration](https://cloudflare-ipfs.com/ipfs/QmahDCSkdED9BLZ3VtH6aJ8P5TmvMYEfA7fJa4hGsvEpi2/).
+
+Once your space is configured you can attach transactions to you proposals via the plugin section:
+
+1. Open the plugin selection
+
+![Open the plugin selection](./snapshot_plugin_section.png)
+
+
+2. Add DAO module plugin
+
+![Add DAO module plugin](./snapshot_add_plugin.png)
+
+3. Add DAO module transaction
+
+<img src="./snapshot_module_add_tx.png"
+     alt="Add DAO module transaction"
+     width="250"/>
+<img src="./snapshot_module_tx_details.png"
+     alt="Enter transactiond etails"
+     width="250" />
+<img src="./snapshot_module_tx_confirm.png"
+     alt="Check transaction details"
+     width="250"/>
+
+4. Check preview of transactions
+
+![Transactions preview](./snapshot_plugin_preview.png)
+
+Once the proposal has been resolved it is possible to use the plugin to submit the proposal to the DAO module via the plugin. 
+
+This can also be done via the hardhat tasks provided in this repository. For more information run `yarn hardhat addProposal --help` or `yarn hardhat executeProposal --help`.
+
+Once the question is available it can be answered via the Realitio web interface (e.g. https://reality.eth.link/app/).
+
+## Monitoring your module
+
+As anyone can submit proposals to your module it is recommended to setup some monitoring. The DAO module relies on the oracle (e.g. Realitio) to provide the correct answer so that no malicious transactions are executed. In the worst case the executor (e.g. the connected Safe) can invalidate a submitted proposal (see [README](../README.md) for more information). 
+
+To make sure that all the involved stakeholders can react in a timely manner, the events emitted by the module contract should be monitored. Each time a new proposal is submitted the contract will emit a `ProposalQuestionCreated` event with the following parameters:
+```
+event ProposalQuestionCreated(
+    bytes32 indexed questionId, // e.g. Realitio question id
+    string indexed proposalId // e.g. Snapshot proposal id
+);
+```
+
+There are different services available for this. E.g. the [OpenZepplin Defender Sentinel](https://docs.openzeppelin.com/defender/sentinel).
```

Comment:
~There are different services available for this. E.g. the~
There are different services available for this such as the